lib/rubocop/cop/rails/index_true.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Check that index: true is not being used in migrations
+      #
+      # @example
+      #   # bad
+      #   add_column :booking_templates, :booking_id, :integer, index: true
+      #
+      #   # good
+      #   add_column :booking_templates, :booking_id, :integer
+      #
+      #   # Remember to add the index separately
+      #   add_index :booking_templates, :booking_id
+      #
+      class IndexTrue < Cop
+        MSG = '`index: true` does not work in an add_column or change_column ' \
+        'statement. Please use `add_index :table, :column`.'.freeze
+
+        def_node_matcher :no_index_true, <<-PATTERN
+          (send nil {:add_column :change_column} _table _column _type (hash $...))
+        PATTERN
+
+        def on_send(node)
+          no_index_true(node) do |args|
+            add_offense(node, :expression, format(MSG)) if index_node(args)
+          end
+        end
+
+        def index_node(args)
+          tuples = args.map(&:children)
```

Comment:
This whole code looks a bit more complex than it needs to be. Why not iterate over the list of arguments, filter out the hashes and check their keys? In general `.source` should rarely be used, as it's super slow. Use names directly from the AST.